src/core/thread/address_resolver.cpp

```diff
@@ -157,9 +181,29 @@ void AddressResolver::InvalidateCacheEntry(Cache &aEntry)
         }
     }
 
+    switch (aEntry.mState)
+    {
+    case Cache::kStateCached:
+        otLogInfoArp(GetInstance(), "Cache entry removed: %s, 0x%04x - %s",
+                     aEntry.mTarget.ToString(stringBuffer, sizeof(stringBuffer)), aEntry.mRloc16,
+                     ConvertInvalidationReasonToString(aReason));
```

Comment:
Is it possible to eliminate `ConvertInvalidationReasonToString `? I guess we could just put the final string literal here since they are not used elsewhere and could be compiled out if logging is not enabled.

Reply:
@bukepo, I am not sure what your suggestion is... 
- In the first version of this PR the method signature was `InvalidateCacheEntry(Cache &aEntry, const char * aReason)`  instead of `InvalidateCacheEntry(Cache &aEntry, InvalidationReason aReason)`. See comments above about this.
- In current version, if the log is not enabled, method `ConvertInvalidationReasonToString()` is not used and should be removed by linker.

Reply:
Got it. Thanks for the explanation.